### PR TITLE
Clean up code mostly based on clippy suggestions

### PR DESCRIPTION
**Pull request by aleksator**

### CODE REVIEW

File: src/cargo/core/compiler/custom_build.rs [hunk 1]
```diff
@@ -208,7 +208,7 @@ fn build_work<'a, 'cfg>(cx: &mut Context<'a, 'cfg>, unit: &Unit<'a>) -> CargoRes
 
     // Be sure to pass along all enabled features for this package, this is the
     // last piece of statically known information that we have.
-    for feat in &unit.features {
+    for &feat in &unit.features {
```
Comment (ehuss) [inline src/cargo/core/compiler/custom_build.rs]: Can you explain these changes to add `&`?  It looks more confusing to me.

Comment (aleksator) [reply src/cargo/core/compiler/custom_build.rs]: The IDE suggested that there's double reference in one of those variables (like &&str), so adding & helps avoiding it. Then I looked for similar patterns and changed them as well.

On the second thought it's not very important, so I have removed that commit.

File: crates/cargo-platform/src/lib.rs [hunk 2]
```diff
@@ -84,16 +84,14 @@ impl Platform {
...
-                        "feature" =>
-                            warnings.push(String::from(
-                                "Found `feature = ...` in `target.'cfg(...)'.dependencies`. \
-                                 This key is not supported for selecting dependencies \
-                                 and will not work as expected. \
-                                 Use the [features] section instead: \
-                                 https://doc.rust-lang.org/cargo/reference/features.html"
-                            )),
-                        _ => (),
+                    Cfg::KeyPair(name, _) => if let "feature" = name.as_str() {
```
[the remark this answers is not in the document]
Comment (aleksator) [reply crates/cargo-platform/src/lib.rs]: It can indeed, changed!

File: crates/cargo-platform/src/lib.rs [hunk 3]
```diff
@@ -84,16 +84,14 @@ impl Platform {
...
-                        "feature" =>
-                            warnings.push(String::from(
-                                "Found `feature = ...` in `target.'cfg(...)'.dependencies`. \
-                                 This key is not supported for selecting dependencies \
-                                 and will not work as expected. \
-                                 Use the [features] section instead: \
-                                 https://doc.rust-lang.org/cargo/reference/features.html"
-                            )),
-                        _ => (),
+                    Cfg::KeyPair(name, _) => if name.as_str() == "feature"{
```
Comment (sinkuu) [inline crates/cargo-platform/src/lib.rs]:
```suggestion
                    Cfg::KeyPair(name, _) => if name.as_str() == "feature" {
```

Comment (aleksator) [reply crates/cargo-platform/src/lib.rs]: Hm, I wonder why doesn't rustfmt catch this. Thank you, I've added the missing space.

**Comment by ehuss** [comment]
Thanks!
@bors r+

**Comment by ehuss** [comment]
@bors retry
E: Failed to fetch https://packages.microsoft.com/ubuntu/16.04/prod/dists/xenial/main/binary-amd64/Packages.bz2  Hash Sum mismatch

**Pull request merged**